src/flash.rs

```diff
@@ -0,0 +1,93 @@
+//! HAL library for Flash Control (FLCTL) - MSP432P401R
+use pac::FLCTL;
+
+/// Typestate for `FlctlConfig` that represents unconfigured FLCTL
+pub struct FlcNotDefined;
+
+/// Typestate for `FlctlConfig` that represents a configured FLCTL
+pub struct FlcDefined;
+
+pub trait State {}
+
+impl State for FlcNotDefined {}
+
+impl State for FlcDefined {}
+
+#[derive(Copy, Clone, PartialEq)]
+pub enum FlWaitSts {
+    _0Ws,
+    _1Ws,
+    _2Ws,
+    _3Ws,
+    _4Ws,
+    _5Ws,
+    _6Ws,
+    _7Ws,
+    _8Ws,
+    _9Ws,
+    _10Ws,
+    _11Ws,
+    _12Ws,
+    _13Ws,
+    _14Ws,
+    _15Ws,
+}
+
+impl FlWaitSts {
+    /// Numerical frequency
+    #[inline(always)]
```

Comment:
Remove if not relevant

https://doc.rust-lang.org/reference/attributes/codegen.html#the-inline-attribute